### PR TITLE
Merge {marshal,unmarshal}TableValue into marshalValue.

### PR DESCRIPTION
**Pull request by petermattis**
This allows easy setting of "typed" values with the standard key/value
methods which will be used shortly for setting and retrieving the
"descriptor ID" value for namespace keys.

### CODE REVIEW

File: client/util.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	case *[]byte:
+		if src.Bytes != nil {
+			*d = src.Bytes
+		} else {
+			*d = nil
+		}
+		return nil
+
+	case *gogoproto.Message:
+		panic("TODO(pmattis): unimplemented")
```
Comment (tschottdorf) [inline client/util.go]: seems awkwardish to panic when you could return an error.

**Comment by tschottdorf** [comment]
LGTM. It's a bit hard to embrace the copious amounts of reflection since we've always tried to avoid it wherever possible, but this being a client library, us having decided to go down this road, and the simplification in this case being significant, this is certainly the right thing to do.

**Comment by petermattis** [comment]
FWIW, the `database/sql` package has similar amounts of reflection internally in order to deal with type conversions in the `{Row,Rows}.Scan()` method. The other bits of reflection to walk over the fields of structs is pretty much required for a Go ORM. The reflection certainly complicates the implementation, but I think it leaves the API usable. I'm very open to alternatives that are both reasonably clean and similarly performant.

**Comment by tschottdorf** [comment]
No need to convince me, I'm on your side.

File: client/util.go (hunk 1 above)
Comment (petermattis) [inline client/util.go]: An error might be ignored or logged. I wanted something more forceful to as a reminder to implement this support. Should be trivial to do, just haven't done it yet.

**Pull request merged**